services/settings-service/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/SettingsServiceDAL.java

```diff
@@ -546,6 +562,57 @@ public AppConfig getAppConfig() {
         return appConfig;
     }
 
+    public void deleteAppConfig() {
+        long startTimeMillis = System.currentTimeMillis();
+        LOGGER.info("SettingsServiceDAL::deleteAppConfig");
+        // NOTE: Could also implement this like deleteTenantSettings by combining SettingsService::REQUIRED_PARAMS
+        // and SettingsService::READ_WRITE_PARAMS and building the Parameter Store path by hand to avoid the call(s)
+        // to getParameters before the call to deleteParameters
+        List<String> parametersToDelete = toSettings(getAppConfig()).stream()
+                .map(s -> toParameterStore(s).name())
+                .collect(Collectors.toList());
+        List<String> batch = new ArrayList<>();
+        try {
+            Iterator<String> it = parametersToDelete.iterator();
+            while (it.hasNext()) {
+                if (batch.size() < 10) {
+                    batch.add(it.next());
+                    // If parametersToDelete % 10 != 0, then be sure to make a request with the remainder
+                    if (!it.hasNext()) {
+                        DeleteParametersResponse response = ssm.deleteParameters(r ->
+                                r.names(batch)
+                        );
+                        LOGGER.info("SettingsServiceDAL::deleteAppConfig removed " + response.deletedParameters().toString());
+                        if (response.hasInvalidParameters() && !response.invalidParameters().isEmpty()) {
+                            LOGGER.warn("SettingsServiceDAL::deleteAppConfig invalid parameters " + response.invalidParameters().toString());
+                        }
+                        // We've reached the end of the request input
+                        break;
+                    }
+                } else {
+                    // Batch has reached max size of 10, make the request
+                    DeleteParametersResponse response = ssm.deleteParameters(r ->
+                            r.names(batch)
+                    );
+                    LOGGER.info("SettingsServiceDAL::deleteAppConfig removed " + response.deletedParameters().toString());
+                    if (response.hasInvalidParameters() && !response.invalidParameters().isEmpty()) {
+                        LOGGER.warn("SettingsServiceDAL::deleteAppConfig invalid parameters " + response.invalidParameters().toString());
```

Comment:
What does it mean about the state of the system if we reach this? Is there some way we can recover the failed deletion?

Reply:
It seemed like a warning vs. an exception to me. It means you probably sent in parameter names that don't exist. The documentation says this collection will be non empty when there are "invalid" parameters, but does not define what invalid means. We log it so that people can have an audit trail if things go wrong. This code branch is a bit pedantic, since the list of parameters to delete is fetch immediately prior to this and I wouldn't expect `getParameters` to return invalid parameters.